pkgs/development/libraries/http-parser/default.nix

```
@@ -12,7 +12,12 @@ stdenv.mkDerivation rec {
   };
 
   NIX_CFLAGS_COMPILE = "-Wno-error";
-  patches = [ ./build-shared.patch ];
+  patches = [ ./build-shared.patch
+              (fetchpatch {
+                # https://github.com/nodejs/http-parser/pull/510
+                url = "https://github.com/nodejs/http-parser/commit/4f15b7d510dc7c6361a26a7c6d2f7c3a17f8d878.patch";
+                sha256 = "sha256-rZZMJeow3V1fTnjadRaRa+xTq3pdhZn/eJ4xjxEDoU4=";
+              })];
```

```suggestion
  patches = [
    ./build-shared.patch
    # https://github.com/nodejs/http-parser/issues/507
    (fetchpatch {
      url = "https://github.com/nodejs/http-parser/commit/4f15b7d510dc7c6361a26a7c6d2f7c3a17f8d878.patch";
      sha256 = "sha256-rZZMJeow3V1fTnjadRaRa+xTq3pdhZn/eJ4xjxEDoU4=";
      })
    ];
```

PR is broken and shows no changes

---

Updated with a381662, sorry for the noise @SuperSandro2000. Accidentally rebased to master before re-rebasing to staging again. Working on a new armv7l build to ensure it's not broken but it's gotta rebuild the world first.

---

@SuperSandro2000 I'm also having trouble getting this to build on staging because of errors with bison, but when I cherry-pick this PR to master:

```
$ nix-build $NIXPKGS --argstr system armv7l-linux -A http-parser
/nix/store/lxszdkz6ji991s38anyss0kl2jllcp8b-http-parser-2.9.4
```

versus building on plain master:
```
$ nix-build $NIXPKGS --argstr system armv7l-linux -A http-parser
this derivation will be built:
  /nix/store/s6n82isd0xynvjzq58q7a8xf7sr0d2cv-http-parser-2.9.4.drv
building '/nix/store/s6n82isd0xynvjzq58q7a8xf7sr0d2cv-http-parser-2.9.4.drv'...
unpacking sources
unpacking source archive /nix/store/wcb9v9vr6fymrjvag8krhlw5ljgg449z-source
source root is source
patching sources
applying patch /nix/store/x3rg29m48h5lkn200mysqz3kr8vq7rz8-build-shared.patch
patching file http_parser.gyp
updateAutotoolsGnuConfigScriptsPhase
configuring
no configure script, doing nothing
building
build flags: SHELL=/nix/store/jr2mir0rp6lf2w88h11lq5apa659pjlp-bash-5.1-p12/bin/bash DESTDIR= PREFIX=\$\(out\) library
gcc -I. -DHTTP_PARSER_STRICT=0  -Wall -Wextra -Werror -O3  -fPIC -c http_parser.c -o libhttp_parser.o
gcc  -shared -Wl,-soname=libhttp_parser.so.2.9 -o libhttp_parser.so.2.9.4 libhttp_parser.o
running tests
check flags: SHELL=/nix/store/jr2mir0rp6lf2w88h11lq5apa659pjlp-bash-5.1-p12/bin/bash DESTDIR= PREFIX=\$\(out\) VERBOSE=y test
gcc -I. -DHTTP_PARSER_STRICT=1  -Wall -Wextra -Werror -O0 -g  -c http_parser.c -o http_parser_g.o
In file included from /nix/store/2ifhclzva25yhffdkwpblmfaycl8w3i3-glibc-2.33-108-dev/include/bits/libc-header-start.h:33,
                 from /nix/store/2ifhclzva25yhffdkwpblmfaycl8w3i3-glibc-2.33-108-dev/include/stdint.h:26,
                 from /nix/store/h1i2dwqwlh9dkniyk2jfillxh1xgdryp-gcc-10.3.0/lib/gcc/armv7l-unknown-linux-gnueabihf/10.3.0/include/stdint.h:9,
                 from http_parser.h:45,
                 from http_parser.c:21:
/nix/store/2ifhclzva25yhffdkwpblmfaycl8w3i3-glibc-2.33-108-dev/include/features.h:397:4: warning: #warning _FORTIFY_SOURCE requires compiling with optimization (-O) [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wcpp-Wcpp8;;]
  397 | #  warning _FORTIFY_SOURCE requires compiling with optimization (-O)
      |    ^~~~~~~
gcc -I. -DHTTP_PARSER_STRICT=1  -Wall -Wextra -Werror -O0 -g  -c test.c -o test_g.o
In file included from /nix/store/2ifhclzva25yhffdkwpblmfaycl8w3i3-glibc-2.33-108-dev/include/bits/libc-header-start.h:33,
                 from /nix/store/2ifhclzva25yhffdkwpblmfaycl8w3i3-glibc-2.33-108-dev/include/stdint.h:26,
                 from /nix/store/h1i2dwqwlh9dkniyk2jfillxh1xgdryp-gcc-10.3.0/lib/gcc/armv7l-unknown-linux-gnueabihf/10.3.0/include/stdint.h:9,
                 from http_parser.h:45,
                 from test.c:21:
/nix/store/2ifhclzva25yhffdkwpblmfaycl8w3i3-glibc-2.33-108-dev/include/features.h:397:4: warning: #warning _FORTIFY_SOURCE requires compiling with optimization (-O) [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wcpp-Wcpp8;;]
  397 | #  warning _FORTIFY_SOURCE requires compiling with optimization (-O)
      |    ^~~~~~~
gcc -Wall -Wextra -Werror -O0 -g   http_parser_g.o test_g.o -o test_g
gcc -I. -DHTTP_PARSER_STRICT=0  -Wall -Wextra -Werror -O3  -c http_parser.c
gcc -I. -DHTTP_PARSER_STRICT=0  -Wall -Wextra -Werror -O3  -c test.c -o test.o
gcc -Wall -Wextra -Werror -O3   http_parser.o test.o -o test_fast
./test_g
http_parser v2.9.4 (0x020904)
sizeof(http_parser) = 32
test_g: test.c:4224: main: Assertion `sizeof(http_parser) == 4 + 4 + 8 + 2 + 2 + 4 + sizeof(void *)' failed.
qemu: uncaught target signal 6 (Aborted) - core dumped
make: *** [Makefile:76: test] Aborted (core dumped)
error: builder for '/nix/store/s6n82isd0xynvjzq58q7a8xf7sr0d2cv-http-parser-2.9.4.drv' failed with exit code 2;
       last 10 log lines:
       > gcc -Wall -Wextra -Werror -O0 -g   http_parser_g.o test_g.o -o test_g
       > gcc -I. -DHTTP_PARSER_STRICT=0  -Wall -Wextra -Werror -O3  -c http_parser.c
       > gcc -I. -DHTTP_PARSER_STRICT=0  -Wall -Wextra -Werror -O3  -c test.c -o test.o
       > gcc -Wall -Wextra -Werror -O3   http_parser.o test.o -o test_fast
       > ./test_g
       > http_parser v2.9.4 (0x020904)
       > sizeof(http_parser) = 32
       > test_g: test.c:4224: main: Assertion `sizeof(http_parser) == 4 + 4 + 8 + 2 + 2 + 4 + sizeof(void *)' failed.
       > qemu: uncaught target signal 6 (Aborted) - core dumped
       > make: *** [Makefile:76: test] Aborted (core dumped)
       For full logs, run 'nix log /nix/store/s6n82isd0xynvjzq58q7a8xf7sr0d2cv-http-parser-2.9.4.drv'.
```